modules/iam-types/src/main/scala/ch/epfl/bluebrain/nexus/commons/iam/types/identity/Identity.scala

```diff
@@ -0,0 +1,72 @@
+package ch.epfl.bluebrain.nexus.commons.iam.types.identity
+
+import akka.http.scaladsl.model.Uri
+import cats.Show
+import io.circe._
+import io.circe.generic.extras._
+import io.circe.generic.extras.semiauto._
+
+import scala.util.Try
+
+/**
+  * Base enumeration type for identity classes.
+  */
+sealed trait Identity extends Product with Serializable
+
+/**
+  * Represents identities that were authenticated from a third party origin.
+  */
+trait Authenticated {
+
+  /**
+    * @return the third party domain from where the authentication origins
+    */
+  def origin: Uri
+}
+
+object Identity {
+
+  /**
+    * The ''user'' identity class.
+    *
+    * @param origin the authentication's origin
+    * @param subject the JWT ''sub'' field
+    */
+  final case class UserRef(origin: Uri, subject: String) extends Identity with Authenticated
+
+  /**
+    * The ''group'' identity class.
+    *
+    * @param origin the authentication's origin
+    * @param group the group name
+    */
+  final case class GroupRef(origin: Uri, group: String) extends Identity with Authenticated
+
+  /**
+    * The ''authenticated'' identity class that represents anyone authenticated from ''origin''.
+    *
+    * @param origin
+    */
+  final case class AuthenticatedRef(origin: Uri) extends Identity with Authenticated
```

Comment:
rename the `origin` to `realm` and define the origin as `Option[Uri]` to be able to express an `AuthenticatedRef` that's not bound to a particular realm.